src/SumatraStartup.cpp

```diff
@@ -796,7 +796,7 @@ int APIENTRY WinMain(_In_ HINSTANCE hInstance, _In_opt_ HINSTANCE hPrevInstance,
 
     retCode = RunMessageLoop();
     SafeCloseHandle(&hMutex);
-    CleanUpThumbnailCache(gFileHistory);

```

Comment:
in fact if we opened 40 file - and now for example can be showed on windows - but FILE_HISTORY_MAX_FREQUENT was 10 ... well, next time the 20 thumbnail doesn't exist!!!!! (we have 40 file in frequently list!! and last 20 file have not thumbnail!!)